### ISSUE TITLE
the level meter needs to be fixed

### ISSUE THREAD

**Issue by multidynamic** [roundware/roundware-android#13]
_eosrei created an issue 2014-12-11_
Halsey reports "my god, the level meter needs to be fixed! this is a long-time annoyance. I’m hoping there is a much better way to handle the metering than we are doing now which is really poor"

:x:  A major issue

_Matthew Adler 2014-12-28_
What is the problem with the meter. Latency and ugly?

_Halsey Burgund REPO OWNER 2014-12-29_
Yes...both latency and the resulting ugliness, but the latency is the real issue, I think. This level meter was a total hack created about 5 years ago, so I'm hoping there is something more standard that we can grab from other audio applications/frameworks that will work better and be smoother. The general look of a left-to-right meter with the discrete blocks of color from green to red is a nice approach, but I'm flexible on the aesthetics. I just want it to be more responsive.
You must have come across level meters in your previous audio app work yes? Anything we can "borrow" or take hints from?

_Matthew Adler 2014-12-29_
Yeah, I have done this before. One of the apps drew a oscilloscope like display. The input just looked at some occasional incoming bytes and processing was quite a bit simpler than this one, but that may make it more responsive.

**Comment by multidynamic** [comment]
Maybe just sampling every 16 ms (instead 100ms) would be less latent. (16ms ~= 60fps)

``` java
private static final int RECORDING_EVENT_INTERVAL_MSEC = 100; // 0.1 sec between updates
```

**Comment by 13rac1** [comment]
A slightly more responsive meter would fix my issues with it. I'd love to see a rebuild/redesign of the activity, but it's Ok for now.

**Comment by hburgund** [comment]
Yes, I agree.  If changing that setting does indeed up the refresh rate, I think that will be fine for now.
